tests/core/go_binary/BUILD.bazel

```diff
@@ -0,0 +1,48 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_test", "go_source")
+
+test_suite(
+    name = "go_binary",
+)
+
+go_binary(
+    name = "hello_plain",
+    srcs = ["hello_world.go"],
+    goarch = "amd64",
+    goos = "windows",
+    pure = "on",
+)
+
+go_binary(
+    name = "hello_no_mode",
+    srcs = ["hello_world.go"],
+    goarch = "amd64",
+    goos = "windows",
+    output_path = "{name}{path}{ext}",
```

Comment:
This would be `out = "hello_no_mode"`. The test can verify that the output file has an .exe extension.